yt_dlp/postprocessor/ffmpeg.py

```diff
@@ -659,12 +661,34 @@ def run(self, info):
 
         temp_filename = prepend_extension(filename, 'temp')
         self.to_screen('Embedding subtitles in "%s"' % filename)
-        self.run_ffmpeg_multiple_files(input_files, temp_filename, opts)
-        os.replace(temp_filename, filename)
+        if info['ext'] in self.AUDIO_EXTS:
+            self.embed_lyrics(input_files)
```

Comment:
Don't use just the filenames. Use the entire `subtitles` dict instead. We have access to language and extension this way.